hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxy.java

```diff
@@ -143,19 +145,19 @@ public Object put(Object key, Object value, long ttl, TimeUnit timeUnit) {
         checkNotNull(key, "key can't be null");
         checkNotNull(value, "value can't be null");
 
-        Object nearCacheKey = toNearCacheKeyWithStrategy(key);
-        try {
-            Data keyData = mapServiceContext.toData(nearCacheKey, partitionStrategy);
-            Object valueBeforeTxn = toObjectIfNeeded(putInternal(keyData, mapServiceContext.toData(value), ttl, timeUnit));
-
-            TxnValueWrapper currentValue = txMap.get(keyData);
-            Type type = valueBeforeTxn == null ? Type.NEW : Type.UPDATED;
-            TxnValueWrapper wrapper = new TxnValueWrapper(value, type);
-            txMap.put(keyData, wrapper);
-            return currentValue == null ? valueBeforeTxn : checkIfRemoved(currentValue);
-        } finally {
```

Comment:
If some exceptions happen in this block, we still call invalidation in the `finally`. We don't do this in the new code. Is it OK if we have an exceptional scenario? It seems in the new implementation we handle failures only on commit.

Reply:
Yes it is ok not to have that finally block. The right place to apply invalidations is commit phase. Depending on the failure or success near cache interaction should happen in there.